include/openmc/position.h

```diff
@@ -221,7 +221,7 @@ namespace fmt {
 template<>
 struct formatter<openmc::Position> : formatter<std::string> {
   template<typename FormatContext>
-  auto format(const openmc::Position& pos, FormatContext& ctx)
+  auto format(const openmc::Position& pos, FormatContext& ctx) const
```

Comment:
```
FAILED: CMakeFiles/libopenmc.dir/src/geometry.cpp.o 
/opt/rh/gcc-toolset-12/root/usr/bin/g++ -DBUILD_TYPE=Release -DCOMPILER_ID=GNU -DCOMPILER_VERSION=12.2.1 -DDAGMC -DEMBREE4 -DFMT_LOCALE -DFMT_SHARED -DGIT_SHA1=\"d0740ce41710f2b9456f5fb2cd8b025fad3a3626\" -DH5_BUILT_AS_DYNAMIC_LIB -DLIBMESH -DNCRYSTAL -DOPENMC_MCPL -DUSE_LIBPNG -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D_LARGEFILE_SOURCE -D_POSIX_C_SOURCE=200809L -Dgsl_CONFIG_DEFAULTS_VERSION=1 -Dlibopenmc_EXPORTS -I/root/openmc/include -I/tmp/tmp11fud5v3/build/include -isystem /usr/include/eigen3 -O3 -DNDEBUG -std=c++17 -fPIC -DH5Oget_info_by_idx_vers=1 -DH5O_info_t_vers=1 -DNDEBUG -pthread -fopenmp -MD -MT CMakeFiles/libopenmc.dir/src/geometry.cpp.o -MF CMakeFiles/libopenmc.dir/src/geometry.cpp.o.d -o CMakeFiles/libopenmc.dir/src/geometry.cpp.o -c /root/openmc/src/geometry.cpp
In file included from /root/openmc/include/openmc/cell.h:19,
                 from /root/openmc/src/geometry.cpp:7:
/root/openmc/include/openmc/position.h: In instantiation of ‘auto fmt::v6::formatter<openmc::Position>::format(const openmc::Position&, FormatContext&) const [with FormatContext = fmt::v6::basic_format_context<std::back_insert_iterator<fmt::v6::internal::buffer<char> >, char>]’:
/usr/include/fmt/core.h:899:28:   required from ‘static void fmt::v6::internal::value<Context>::format_custom_arg(const void*, typename Context::parse_context_type&, Context&) [with T = openmc::Position; Formatter = fmt::v6::formatter<openmc::Position>; Context = fmt::v6::basic_format_context<std::back_insert_iterator<fmt::v6::internal::buffer<char> >, char>; typename Context::parse_context_type = fmt::v6::basic_format_parse_context<char>]’
/usr/include/fmt/core.h:883:19:   required from ‘fmt::v6::internal::value<Context>::value(const T&) [with T = openmc::Position; Context = fmt::v6::basic_format_context<std::back_insert_iterator<fmt::v6::internal::buffer<char> >, char>]’
/usr/include/fmt/core.h:1217:39:   required from ‘fmt::v6::internal::value<Context> fmt::v6::internal::make_arg(const T&) [with bool IS_PACKED = true; Context = fmt::v6::basic_format_context<std::back_insert_iterator<buffer<char> >, char>; type <anonymous> = fmt::v6::internal::type::custom_type; T = openmc::Position; typename std::enable_if<IS_PACKED, int>::type <anonymous> = 0]’
/usr/include/fmt/core.h:1356:66:   required from ‘fmt::v6::format_arg_store<Context, Args>::format_arg_store(const Args& ...) [with Context = fmt::v6::basic_format_context<std::back_insert_iterator<fmt::v6::internal::buffer<char> >, char>; Args = {int, int, int, int, openmc::Position}]’
/usr/include/fmt/core.h:1656:18:   required from ‘fmt::v6::format_arg_store<fmt::v6::basic_format_context<std::back_insert_iterator<fmt::v6::internal::buffer<Char> >, Char>, typename std::remove_reference<Args>::type ...> fmt::v6::internal::make_args_checked(const S&, fmt::v6::remove_reference_t<Args>& ...) [with Args = {int&, int&, int&, int&, openmc::Position&}; S = char [59]; Char = char]’
/usr/include/fmt/core.h:1747:43:   required from ‘std::__cxx11::basic_string<Char> fmt::v6::format(const S&, Args&& ...) [with S = char [59]; Args = {int&, int&, int&, int&, openmc::Position&}; Char = char]’
/root/openmc/src/geometry.cpp:306:18:   required from here
/root/openmc/include/openmc/position.h:226:42: error: passing ‘const fmt::v6::formatter<openmc::Position>’ as ‘this’ argument discards qualifiers [-fpermissive]
  226 |     return formatter<std::string>::format(
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  227 |       fmt::format("({}, {}, {})", pos.x, pos.y, pos.z), ctx);
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/fmt/ostream.h:13,
                 from /root/openmc/src/geometry.cpp:4:
```

I am encountering the following error when building OpenMC with `fmt-devel-6.2.1-1` on Manilinux. Is there a way to support older versions as well?

Reply:
If you have a proposed solution that works for current releases as well as older releases, I am happy to entertain a PR. I'll note however that version 6.2.1 is more than 4 years old at this point.